hazelcast/src/main/java/com/hazelcast/client/impl/protocol/util/SafeBuffer.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.impl.protocol.util;
+
+import com.hazelcast.nio.Bits;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.charset.Charset;
+
+public class SafeBuffer implements ClientProtocolBuffer {
```

Comment:
Can you add some javaDoc for `SafeBuffer`? Currently it is not clear why this buffer is safe and under what conditions this buffer should be used.

Also can you add some tests for `SafeBuffer`?